app/controllers/users_controller.rb

```diff
@@ -15,15 +13,17 @@ class UsersController < ApplicationController
     if current_user
       redirect_to({id: current_user.id}, alert: t('common.errors.not_found_error'))
     else
-      redirect_to root_path, alert: t('common.errors.not_authorized')
+      redirect_to root_path, alert: t('common.errors.not_found_error')
```

Comment:
Previously, we somewhat decided by design to use `not_authorized` here. The intention was to provide as few information as possible to those who are just trying to access invalid resources. Then, just trying around with different URLs won't give you any clue whether the user exists or not. However, this mechanism wasn't working here correctly, since any user logged in would receive an 404. Ideally, we would only show admins a 404 (and otherwise allow access), and all other users (logged in or anonymous) just receive a "generic" error. Those users logged in should be, that was the original idea at that time, redirect to their own profile instead.

Reply:
It should work as you described now.